common_benchmark_suite/openxla/benchmark/comparative_suite/tflite/model_definitions.py

```diff
@@ -0,0 +1,181 @@
+    name=utils.SEQ_LEN_NAME("BERT_BASE_FP16_TFLITE_I32"),
+    tags=[utils.SEQ_LEN_TAG],
+    model_impl=TFLITE_MODEL_IMPL,
+    model_parameters={
+        "model_uri":
+            string.Template(
+                JAX_MODELS_GCS_DIR +
+                "/BERT_BASE_FP32_JAX_I32_SEQLEN${seq_len}/model_fp16.tflite"),
+    },
+    artifacts_dir_url=ARTIFACTS_DIR_URL_TEMPLATE,
+    exported_model_types=[
+        def_types.ModelArtifactType.TFLITE_FP32,
+        def_types.ModelArtifactType.TOSA_MLIR,
+    ],
+)
+
+BERT_BASE_FP16_TFLITE_I32_INPUT_SEQUENCES = utils.build_input_sequence_models(
+    template=BERT_BASE_FP16_TFLITE_I32_INPUT_SEQUENCE_TEMPLATE,
+    input_sequence_lengths=[8, 32, 64, 128, 256, 512])
+
+BERT_BASE_DYN_QUANT_TFLITE_I32_INPUT_SEQUENCE_TEMPLATE = utils.ModelTemplate(
+    name=utils.SEQ_LEN_NAME("BERT_BASE_DYN_QUANT_TFLITE_I32"),
+    tags=[utils.SEQ_LEN_TAG],
+    model_impl=TFLITE_MODEL_IMPL,
+    model_parameters={
+        "model_uri":
+            string.Template(
+                JAX_MODELS_GCS_DIR +
+                "/BERT_BASE_FP32_JAX_I32_SEQLEN${seq_len}/model_dynamic_range_quant.tflite"
+            ),
+    },
+    artifacts_dir_url=ARTIFACTS_DIR_URL_TEMPLATE,
+    exported_model_types=[
+        def_types.ModelArtifactType.TFLITE_DYNAMIC_RANGE_QUANT,
+        def_types.ModelArtifactType.TOSA_MLIR,
+    ],
+)
+
+BERT_BASE_DYN_QUANT_TFLITE_I32_INPUT_SEQUENCES = utils.build_input_sequence_models(
+    template=BERT_BASE_DYN_QUANT_TFLITE_I32_INPUT_SEQUENCE_TEMPLATE,
+    input_sequence_lengths=[8, 32, 64, 128, 256, 512])
+
+BERT_BASE_INT8_TFLITE_I32_INPUT_SEQUENCE_TEMPLATE = utils.ModelTemplate(
+    name=utils.SEQ_LEN_NAME("BERT_BASE_INT8_TFLITE_I32"),
+    tags=[utils.SEQ_LEN_TAG],
+    model_impl=TFLITE_MODEL_IMPL,
+    model_parameters={
+        "model_uri":
+            string.Template(
+                JAX_MODELS_GCS_DIR +
+                "/BERT_BASE_FP32_JAX_I32_SEQLEN${seq_len}/model_int8.tflite"),
+    },
+    artifacts_dir_url=ARTIFACTS_DIR_URL_TEMPLATE,
+    exported_model_types=[
+        def_types.ModelArtifactType.TFLITE_INT8,
+        def_types.ModelArtifactType.TOSA_MLIR,
+    ],
+)
+
+BERT_BASE_INT8_TFLITE_I32_INPUT_SEQUENCES = utils.build_input_sequence_models(
+    template=BERT_BASE_INT8_TFLITE_I32_INPUT_SEQUENCE_TEMPLATE,
+    input_sequence_lengths=[8, 32, 64, 128, 256, 512])
+
+VIT_CLASSIFICATION_FP32_TFLITE_3X224X224XF32 = def_types.Model(
+    name="VIT_CLASSIFICATION_FP32_TFLITE_3X224X224XF32",
+    tags=["fp32", "batch-1"],
+    model_impl=TFLITE_MODEL_IMPL,
+    model_parameters={
+        "model_uri":
+            f"{JAX_MODELS_GCS_DIR}/VIT_CLASSIFICATION_JAX_3X224X224XF32/model_fp32.tflite",
+    },
+    artifacts_dir_url=ARTIFACTS_DIR_URL_TEMPLATE,
+    exported_model_types=[
+        def_types.ModelArtifactType.TFLITE_FP32,
+        def_types.ModelArtifactType.TOSA_MLIR,
+    ],
+)
+
+VIT_CLASSIFICATION_FP16_TFLITE_3X224X224XF32 = def_types.Model(
```

Comment:
What is the difference between this and the one defined on JAX models? https://github.com/openxla/openxla-benchmark/blob/main/common_benchmark_suite/openxla/benchmark/comparative_suite/jax/model_definitions.py#L526

Reply:
It uses the `def_types.ModelArtifactType.TFLITE_FP16` artifact defined in `VIT_CLASSIFICATION_JAX_3X224X224XF32` as input to `iree-import-tflite` to generate a `def_types.ModelArtifactType.TOSA_MLIR` artifact. So there is a dependency chain but we create our own TFLite model definitions so that they include TOSA mlir artifacts.